core/src/main/scala/cats/Semigroupal.scala

```diff
@@ -0,0 +1,16 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+ * [[Semigroupal]] generalizes [[Functor]] by allowing functions of any
+ * arity to be applied to an effectful value. It encodes this by pairing
+ * together two effectful values (which can then be extended to be three
+ * or more) within a context, which can then be mapped over via the [[Functor]]
+ * instance.
+ */
+@typeclass trait Semigroupal[F[_]] extends Functor[F] {
```

Comment:
I think that’s not a good idea to restrict `Semigroupal` to be a specialization of `Functor`. For instance, you can not anymore implement [Monoidal[Show]](https://github.com/non/cats/pull/555#issuecomment-145351867).